### PR TITLE
Task-40023 : A user can markAsRead or hide a notification which not b…

### PR DESCRIPTION
**Pull request by rdenarie**
…elong to him

This fix add a check on the receiver of the web notification. If the current user is not the receiver, he cannot mark as read or hide

### CODE REVIEW

File: commons-api/src/main/java/org/exoplatform/commons/api/notification/rest/WebNotificationRestService.java [hunk 1]
```diff
@@ -96,15 +97,27 @@ public Response updateNotifications(@ApiParam(value = "operation", required = tr
         if (notificationId == null) {
           return Response.status(Response.Status.BAD_REQUEST).build();
         } else {
-          webNftService.markRead(notificationId);
+          NotificationInfo notification = webNftService.getNotificationInfo(notificationId);
+          if (currentUser.equals(notification.getTo())) {
+            webNftService.markRead(notificationId);
+          } else {
+            return Response.status(Response.Status.UNAUTHORIZED).build();
```
Comment (boubaker) [inline commons-api/src/main/java/org/exoplatform/commons/api/notification/rest/WebNotificationRestService.java:105]: could you please add a `log.warn` entry to notice this to Ops (This shouldn't happen at all but when it happens, this means whether it's a real tentative of attack or the product has a bug). Same as line 118.

( BTW, could you please format all added classes using [IDE Formatter](http://developer.exoplatform.org/#id-formatter-settings) )

Comment (rdenarie) [reply commons-api/src/main/java/org/exoplatform/commons/api/notification/rest/WebNotificationRestService.java:105]: Ok for the first point I'll do it.

For the formater, my intellij is configured in accordance with the link you provide.
Can you explain what is the problem with the new class  ?

Comment (boubaker) [reply commons-api/src/main/java/org/exoplatform/commons/api/notification/rest/WebNotificationRestService.java:105]: Nothing serious, multiple new lines at the end of the file, no newline between methods ( see WebNotificationRestServiceTest.java )

Comment (rdenarie) [reply commons-api/src/main/java/org/exoplatform/commons/api/notification/rest/WebNotificationRestService.java:105]: Does the formatter should remove new lines add the end of the file ? Does it add new lines between methods  ?
What about in your configuration ?

Comment (rdenarie) [reply commons-api/src/main/java/org/exoplatform/commons/api/notification/rest/WebNotificationRestService.java:105]: In fact, the fomatter do that, but I need to ask him explicitly to format. 
Done !

**Review by boubaker — APPROVED** [review]

**Review by ahamdi — APPROVED** [review]

**Pull request merged**